### PR TITLE
chore: Update CircleCi config to use parallel connector jobs

### PR DESCRIPTION
**Pull request by christophd**
PR uses parallel jobs for connectors on CircleCi. 

- connector-support modules are built first in sequential order
- all connectors are built in parallel to each other
- connector-support-catalog is built at the very end to collect all connectors
- nightly job still uses "old" way of running all connectors with single job

Relates to #5797

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -513,17 +560,260 @@ workflows:
...
+            - telegram
+            - timer
+            - twitter
+            - servicenow
+            - webhook
+            - concur
+            - kudu
+            - odata
+            - knative
+            - flow
```
Comment (squakez) [inline .circleci/config.yml]: This is a bit ugly to see... too much duplication. Isn't there any way to group those and reference with a unique label instead?

Comment (christophd) [reply .circleci/config.yml]: agreed, but yet I have not found a better way to do it

**Comment by christophd** [comment]
the config seems to have some issue because CircleCi complains about it - Darn! YAML!

File: .circleci/config.yml (hunk 1 above)
Comment (squakez) [reply .circleci/config.yml]: Maybe someone with experience on the tool can tell (@zregvart ?)

File: .circleci/config.yml [hunk 2]
```diff
@@ -162,31 +162,78 @@ jobs:
...
       <<: *common_env
     steps:
       - checkout
       - restore_cache:
-          key: syndesis-mvn-connector-{{ checksum "app/connector/pom.xml" }}
+          key: syndesis-mvn-connector-{{ checksum "app/connector-support/pom.xml" }}
       - <<: *load_m2
       - run:
-          name: Build Connectors
+          name: Build Connector Util
```
Comment (zregvart) [inline .circleci/config.yml]: Perhaps we can introduce two modules one containing all support modules for connectors and one containing only connectors. Would make these bits a bit simpler.

File: .circleci/config.yml [hunk 3]
```diff
@@ -162,31 +162,78 @@ jobs:
...
           path: build_log.txt
       - save_cache:
-          key: syndesis-mvn-connector-{{ checksum "app/connector/pom.xml" }}
+          key: syndesis-mvn-connector-{{ checksum "app/connector-support/pom.xml" }}
           paths:
           - ~/.m2
       - <<: *save_m2
 
+  # Connector parameterized depends on connector-base, mount workspace .m2
+  connector:
```
Comment (zregvart) [inline .circleci/config.yml]: Here `connector` is a job, I think it needs to be a [command](https://circleci.com/docs/2.0/reusing-config/#using-the-parameters-declaration)

Comment (christophd) [reply .circleci/config.yml]: I wanted to have a parameterized job to call it from the workflow: https://circleci.com/docs/2.0/reusing-config/#authoring-parameterized-jobs

File: .circleci/config.yml (hunk 2 above)
Comment (christophd) [reply .circleci/config.yml]: I have improved a bit to run all modules in one run command. Also it is a bit more complicated than I thought. We need to build some support modules first then all connectors and at the very end the connector-catalog.

**Comment by christophd** [comment]
> the config seems to have some issue because CircleCi complains about it - Darn! YAML!

YAML has not been the problem. I should have read the documentation of parameterized jobs carefully: `Note: Invoking jobs multiple times in a single workflow and parameters in jobs are available in configuration version 2.1 and later.`

So we are using version 2 at the moment. Any objections to using version 2.1?

File: .circleci/config.yml (hunk 3 above)
Comment (zregvart) [reply .circleci/config.yml]: Oh cool.

File: .circleci/config.yml [hunk 4]
```diff
@@ -509,20 +560,232 @@ workflows:
...
+            - connector-webhook
+            - connector-concur
+            - connector-kudu
+            - connector-odata
+            - connector-knative
+            - connector-flow
       - meta:
           requires:
-            - connector
+            - connector-catalog
```
Comment (squakez) [inline .circleci/config.yml]: Tidier now!

**Comment by christophd** [comment]
so the build config is still failing with version 2.1 set. I saw something on the circleci Q&A which might be the case here, too:

> I strongly suspect the issue here is you don’t have Build Processing enabled in your project settings (under “Advanced” on your project’s settings page). Your original config runs fine for me when I have that on. The bug in this case may be that we need a much better error handling when passing 2.1

I do not see the project settings nor can I adjust these @zregvart can you?

**Comment by zregvart** [comment]
@christophd found [the setting](https://circleci.com/gh/syndesisio/syndesis/edit#advanced-settings) and turned 2.1 support to on.

**Comment by zregvart** [comment]
We have a [build error](https://circleci.com/gh/syndesisio/syndesis/90629) now, something with YAML.

**Comment by christophd** [comment]
@zregvart @squakez I have finally sorted out all problems with the parallel connector build and the workflow is running successfully.

I have removed WIP label now can you two please have another look?

We still need to see if this all brings us a lot of speed in the build jobs. 

This one here with parallel jobs took 39 minutes: 
https://circleci.com/workflow-run/923c5821-09a4-41cd-9f84-6790e7762151

This one with sequential jobs took 45 minutes:
https://circleci.com/workflow-run/2321aa81-0252-431f-b90c-cbb7bd498f93

This one with sequential jobs took 33 minutes:
https://circleci.com/workflow-run/484860c9-0d14-4fdb-b14e-29b2a05c0423

**Comment by squakez** [comment]
As you've already mentioned, the parallelism is a good thing if there are enough resources to really run in parallel, otherwise it ends up in a sequential execution anyhow. Even if there is not yet any big gain it is a nice example to see how parallel builds work, so we can leverage that for the future. Nice work!

**Pull request merged**